### ISSUE TITLE
Wordpress 4.9.7 and Group assign of 'Registered'

### ISSUE THREAD

**Issue by donovandb** [onelogin/wordpress-saml#69]
Another EDIT: It may be that this updated Plug-in is messing with the Groups plugin feature of automatically adding a group upon creating a new user.  Any idea if there is code updates here between this version and version 2.5.0 that may have changed the behavior when creating an user?

EDIT:  Sorry, I think this has to do with the WordPress Plugin 'Groups'. I've used that in the past along with SSO and this plug-in and it has always added 'Registered' as the default group upon signing in. With WordPress 4.9.7 and this latest version of onelogin wordpress SAML, it does not do that anymore.

Hello, on a new integration, Successfully created wordpress users are no longer being assigned 'Registered' in the group with WordPress 4.9.7.  

Any ideas?

**Comment by donovandb** [comment]
@pitbulk, thanks for the reply. 

I believe 'role' and 'group' are different (When the wordpress Groups plug-in is installed).  Role is getting propagated correctly. Not groups.

I'll post this anyway.. as I still can't find where the issue is coming from:
```
[root@vps2 wp-content]# pwd
/home/cessnaowner/public_html/wp-content
[root@vps2 wp-content]# grep -Hinre 'default_role' *
plugins/better-wp-security/core/lib/class-itsec-lib-password-requirements.php:154:                      $canonical = ITSEC_Lib_Canonical_Roles::get_canonical_roleubscriber' ) );
plugins/onelogin-saml-sso/php/functions.php:248:                        $userdata['role'] = get_option('default_role');
plugins/wp-security-audit-log/classes/Sensors/System.php:522:           if ( $is_option_page && wp_verify_nonce( $post_array['_wpnonce'], 'general-options' ) && !
plugins/wp-security-audit-log/classes/Sensors/System.php:523:                   $old = get_option( 'default_role' );
plugins/wp-security-audit-log/classes/Sensors/System.php:524:                   $new = trim( $post_array['default_role'] );
plugins/wp-security-audit-log/classes/Sensors/System.php:605:           if ( isset( $post_array['action'] ) && 'update' === $post_array['action'] && isset( $post_
plugins/wp-security-audit-log/classes/Sensors/System.php:606:                   $old_role = get_option( '_bbp_default_role' );
plugins/wp-security-audit-log/classes/Sensors/System.php:607:                   $new_role = $post_array['_bbp_default_role'];
```

**Comment by donovandb** [comment]
I did some further testing. I found another WordPress 4.9.7 site that we have and installed the 'Groups' plugin there too. I then simply turned on the local WordPress membership registration, registered (/wp-login.php?action=register) a user, and then went to the users panel. 'Registered' was indeed applied to this manually created user, which is a feature of 'Groups' (it automatically assigns 'Registered' to all registered users)

Next, I installed the OneLogin SAML SSO plug-in, configured it, then signed in using Single Sign On. This verified the same error of 'Registered' not automatically being applied as noted here. 

Finally, I updated WordPress to 4.9.8, just to check if there was something in that update that might fix the issue. It did not.

I can only concluded that there is likely some incompatibility between the Groups and OneLogin SAML SSO plugins, that is preventing Groups from assigning 'Registered' to newly created users using the OneLogin SAML SSO plugin. Again, this works on older versions of the two plugins.
